code/ui/manager/src/components/mobile/navigation/MobileNavigation.tsx

```diff
@@ -20,14 +20,12 @@ const useFullStoryName = () => {
   const { index } = useStorybookState();
   const currentStory = useStorybookApi().getCurrentStoryData();
 
-  if (!currentStory) {
-    return '';
-  }
+  if (!currentStory) return '';
 
   let fullStoryName = currentStory.renderLabel?.(currentStory) || currentStory.name;
   let node = index[currentStory.id];
 
-  while ('parent' in node && node.parent && index[node.parent]) {
+  while ('parent' in node && node.parent && index[node.parent] && fullStoryName.length < 24) {
```

Comment:
this won't make it stop at 24 characters, it will just make sure that it doesn't add more parents when it has _exceeded_ 24 characters.

Take this example:

```
Long Root Parent Name/Loooooooong Name/Medium Name/Short Name
  ^21 char           ^    ^16 char    ^  ^11 char ^  ^10 char
```

Will result in:

```
Loooooooong Name/Medium Name/Short Name
    ^16 char    ^  ^11 char ^   ^10 char
= 39 characters
```

Which is still better than nothing, but I just wanted to make sure we all understood this.
If we wanted this to always stop at 24 characters, we would afterwards remove any characters before the last 24 with a substr or similar, which would result in:

```
e/Medium Name/Short Name
```

Reply:
Thanks @JReinhold. We don't want to stop exactly at 24 characters. This is just an arbitrary number based on the tests I made to get a sense as to when we can put the parent or not. If the parent has a long name for example, we will use the  css truncate with "..." to actually truncate the text.